--- git status ---
On branch devel
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   source/robot_lab/robot_lab/tasks/locomotion/velocity/config/quadruped/unitree_a1/rough_env_cfg.py
	modified:   source/robot_lab/robot_lab/tasks/locomotion/velocity/velocity_env_cfg.py

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/source/robot_lab/robot_lab/tasks/locomotion/velocity/config/quadruped/unitree_a1/rough_env_cfg.py b/source/robot_lab/robot_lab/tasks/locomotion/velocity/config/quadruped/unitree_a1/rough_env_cfg.py
index e92769e..5e0eb20 100644
--- a/source/robot_lab/robot_lab/tasks/locomotion/velocity/config/quadruped/unitree_a1/rough_env_cfg.py
+++ b/source/robot_lab/robot_lab/tasks/locomotion/velocity/config/quadruped/unitree_a1/rough_env_cfg.py
@@ -146,9 +146,9 @@ class UnitreeA1RoughEnvCfg(LocomotionVelocityRoughEnvCfg):
         self.rewards.upward.weight = 3.0
 
         # skate rewards
-        self.rewards.skate_distance_penalty.weight = 1.0
-        self.rewards.feet_skate_contact.weight = 1.0
-        self.rewards.skate_rot_penalty.weight = 1.0
+        self.rewards.skate_distance_penalty.weight = 0
+        self.rewards.feet_skate_contact.weight = 0
+        self.rewards.skate_rot_penalty.weight = 0
 
         # If the weight of rewards is 0, set rewards to None
         if self.__class__.__name__ == "UnitreeA1RoughEnvCfg":
diff --git a/source/robot_lab/robot_lab/tasks/locomotion/velocity/velocity_env_cfg.py b/source/robot_lab/robot_lab/tasks/locomotion/velocity/velocity_env_cfg.py
index e89ff4c..db6ff25 100644
--- a/source/robot_lab/robot_lab/tasks/locomotion/velocity/velocity_env_cfg.py
+++ b/source/robot_lab/robot_lab/tasks/locomotion/velocity/velocity_env_cfg.py
@@ -109,8 +109,8 @@ class MySceneCfg(InteractiveSceneCfg):
             activate_contact_sensors=False,
         ),
         init_state=ArticulationCfg.InitialStateCfg(
-            pos=(1.0, -0.39, 0.0),
-            # pos=(0.0, 0.0, 0.0),
+            # pos=(1.0, -0.39, 0.0),
+            pos=(2.0, 2.0, 0.0),
             rot=(1.0, 0.0, 0.0, 0.0),
             joint_pos={  # Указываем реальные имена шарниров из ski.usd
                 "trj0": 0.0,
@@ -304,6 +304,17 @@ class ObservationsCfg:
             clip=(-1.0, 1.0),
             scale=1.0,
         )
+        skate_pos_rel = ObsTerm(
+            func=mdp.skate_pos_rel,
+            noise=Unoise(n_min=-0.1, n_max=0.1),
+            scale=1.0,
+        )
+
+        skate_rot_rel = ObsTerm(
+            func=mdp.skate_rot_rel,
+            noise=Unoise(n_min=-0.1, n_max=0.1),
+            scale=1.0,
+        )
 
         def __post_init__(self):
             self.enable_corruption = False
@@ -744,7 +755,7 @@ class LocomotionVelocityRoughEnvCfg(ManagerBasedRLEnvCfg):
     """Configuration for the locomotion velocity-tracking environment."""
 
     # Scene settings
-    scene: MySceneCfg = MySceneCfg(num_envs=1, env_spacing=2.5)
+    scene: MySceneCfg = MySceneCfg(num_envs=4096, env_spacing=2.5)
     # Basic settings
     observations: ObservationsCfg = ObservationsCfg()
     actions: ActionsCfg = ActionsCfg()